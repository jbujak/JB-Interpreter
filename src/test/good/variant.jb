variant {
	int num;
	string str;
} my_variant;

int main() {
	my_variant var1 = :num(1);
	my_variant var2 = :str("a");

	string s = "";
	s = case var1 of
		| :num(n) -> ":num(" + int_to_string(n) + ")"
		| :str(s) -> (":str(" + s + ")");
	print(s);

	s = case var2 of
		| :num(n) -> ":num(" + int_to_string(n) + ")"
		| :str(s) -> ":str(" + s + ")";
	print(s);
}
