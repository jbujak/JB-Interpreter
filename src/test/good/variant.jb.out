:num(1)
:str(a)
